src/biogeophys/HillslopeHydrologyMod.F90

```diff
@@ -0,0 +1,1326 @@
+          patch%wtgcell(col%patchi(c):col%patchf(c)) = 0._r8
+
+          ! assumes trees are 1-8, shrubs 9-11, and grasses 12-14
+          ! and puts the lowest ivt on the lowland column
+          if ((patch%itype(pdom(1)) > patch%itype(psubdom(1)) &
+               .and. patch%itype(psubdom(1)) > 0) .or. &
+               (patch%itype(pdom(1)) == 0)) then
+             plow = psubdom(1)
+             phigh = pdom(1)
+          else
+             plow = pdom(1)
+             phigh = psubdom(1)
+          endif
+
+          ! Special cases (subjective)
+
+          ! if NET/BDT assign BDT to lowland
+          if ((patch%itype(pdom(1)) == ndllf_evr_tmp_tree) .and. pftcon%is_tree(patch%itype(psubdom(1)))) then
+             plow = psubdom(1)        
+             phigh = pdom(1)
+          endif
+          ! if C3/C4 assign C4 to lowland
+          if ((patch%itype(pdom(1)) == nc4_grass) .and. (patch%itype(psubdom(1)) == nc3_nonarctic_grass)) then
+             plow = pdom(1)        
+             phigh = psubdom(1)
+          endif
+          if ((patch%itype(pdom(1)) == nc3_nonarctic_grass) .and. (patch%itype(psubdom(1)) == nc4_grass)) then
+             plow = psubdom(1)        
+             phigh = pdom(1)
+          endif
+
+          if(col%cold(c) == ispval) then
+             ! lowland column
+             patch%wtcol(plow)   = sum_wtcol
+             patch%wtlunit(plow) = sum_wtlun
+             patch%wtgcell(plow) = sum_wtgrc
+          else
+             ! upland columns
+             patch%wtcol(phigh)   = sum_wtcol
+             patch%wtlunit(phigh) = sum_wtlun
+             patch%wtgcell(phigh) = sum_wtgrc
+          endif
+       endif
+    enddo    ! end loop c
+
+  end subroutine HillslopeDominantLowlandPft
+
+  !------------------------------------------------------------------------
+  subroutine HillslopePftFromFile(bounds,col_pftndx)
+    !
+    ! !DESCRIPTION: 
+    ! Reassign patch type using indices from surface data file
+    ! Assumes one patch per hillslope column
+    ! In preparation for this reassignment of patch type, only the
+    ! first patch was given a non-zero weight in surfrd_hillslope.
+    !
+    ! !USES
+    use ColumnType      , only : col                
+    use PatchType       , only : patch
+
+    !
+    ! !ARGUMENTS:
+    type(bounds_type), intent(in) :: bounds
+    integer, intent(in)           :: col_pftndx(:)
+    !
+    ! !LOCAL VARIABLES:
+    integer :: p,c    ! indices
+    integer :: npatches_per_column
+    logical :: check_npatches = .true.
+    
+    !------------------------------------------------------------------------
+
+    do c = bounds%begc, bounds%endc
+       if (col%is_hillslope_column(c)) then
+          ! In preparation for this re-weighting of patch type
+          ! only first patch was given a non-zero weight in surfrd_hillslope
+          npatches_per_column = 0
+          do p = col%patchi(c), col%patchf(c)
+             patch%itype(p) = col_pftndx(c)
```

Comment:
- [ ] As you did in HillslopeSetLowlandUplandPfts, there should also be an update of `patch%mxy` here.